rust/src/client.rs

```diff
@@ -89,6 +111,119 @@ impl Session {
 struct SessionManager {
     logger: Logger,
     tx: tokio::sync::mpsc::Sender<command::Command>,
+    selector: QueueSelector,
```

Comment:
`QueueSelector` makes sense to Producer only. Best to move to the Producer struct.